src/functionalTest/java/uk/gov/hmcts/reform/cdapi/CommonDataApiFunctionalTest.java

```diff
@@ -304,5 +310,39 @@ void shouldReturnSuccessWithServiceIdNoChilds() {
         }
     }
 
+    @Test
+    @ToggleEnable(mapKey = MAP_KEY_LOV, withFeature = true)
+    @ExtendWith(FeatureToggleConditionExtension.class)
+    void shouldReturnSuccess_Valid_ServiceID() {
+        Response response =
+            commonDataApiClient.retrieveCategoriesByCategoryIdSuccess(PATH_LOV, PARAM_LISTING_STATUS_WITH_SERVICE_ID);
+        if (OK.value() == response.getStatusCode()) {
+            var categories = response.getBody().as(Categories.class);
+            assertNotNull(categories);
+            assertThat(categories.getListOfCategory()).hasSizeGreaterThan(1);
+            categories.getListOfCategory().forEach(h -> assertEquals("ListingStatus", h.getCategoryKey()));
+        } else {
+            assertEquals(NOT_FOUND.value(), response.getStatusCode());
+        }
+    }
+
+    @Test
+    @ToggleEnable(mapKey = MAP_KEY_LOV, withFeature = true)
+    @ExtendWith(FeatureToggleConditionExtension.class)
+    void shouldReturnSuccess_Empty_ServiceID() {
+        Response response =
+            commonDataApiClient.retrieveCategoriesByCategoryIdSuccess(
+                PATH_LOV,
+                PARAM_LISTING_STATUS_WITH_EMPTY_SERVICE_ID
+            );
+        if (OK.value() == response.getStatusCode()) {
+            var categories = response.getBody().as(Categories.class);
+            assertNotNull(categories);
+            assertThat(categories.getListOfCategory()).hasSizeGreaterThan(1);
+            categories.getListOfCategory().forEach(h -> assertEquals("ListingStatus", h.getCategoryKey()));
+        } else {
+            assertEquals(NOT_FOUND.value(), response.getStatusCode());
+        }
+    }
```

Comment:
these 2 functional test cases are partially covered, as your test case checking for either response can be NOT_FOUND or OK , can you pls add integration test cases to prove that your change will fetch exact number of records which have empty service id and assert their count and content.  
For this you have to insert couple records with empty service id in db and one record with child records, then call this endpoint with empty service id for that category key. 
Then check one record has child nodes and another one does not have.

Reply:
1. these 2 functional test cases are partially covered, as your test case checking for either response can be NOT_FOUND or OK, can you pls add integration test cases to prove that your change will fetch exact number of records which have empty service id and assert their count and content- 
@rajkatla-hmcts : Is data dynamic or permanent?

2. For this you have to insert couple records with empty service id in db and one record with child records, then call this endpoint with empty service id for that category key. - 
@rajkatla-hmcts - This is covered with categoryKey = ListingStatus

3. Then check one record has child nodes and another one does not have.
@rajkatla-hmcts - I did not do any changes in child node functionality.